app/models/entity.rb

```diff
@@ -0,0 +1,122 @@
+      # client.get_updated('Account', last_synchronization.updated_at, Time.now)
+      # client.query('select Id, Name from Account ORDER BY Name')
+    # else
+      fields = self.mapping.values.join(', ')
+      entities = client.query("select Id, LastModifiedDate, #{fields} from #{self.external_entity_name} ORDER BY LastModifiedDate DESC")
+      entities = entities.to_a
+
+      index = 0
+      entities.each_with_index do |entity, i|
+        if entity.LastModifiedDate < last_synchronization.updated_at
+          index = i - 1
+          break
+        end
+      end
+      index > 0 ? entities[0..index] : []
+    # end
+  end
+
+  def push_entities_to_external(external_client, connec_entities, organization)
+    connec_entities.each do |connec_entity|
+      idmap = IdMap.find_or_create_by(connec_id: connec_entity['id'], connec_entity: self.connec_entity_name.downcase, organization_id: organization.id)
+      # Entity does not exist in external
+      if idmap.salesforce_id.blank?
+        external_id = self.create_entity_to_external(external_client, connec_entity)
+        idmap.update_attributes(salesforce_id: external_id, salesforce_entity: self.external_entity_name)
+      else
+        self.update_entity_to_external(external_client, connec_entity, idmap.salesforce_id)
+      end
+    end
+  end
+
+  def create_entity_to_external(client, connec_entity)
+    client.create(self.external_entity_name, data_to_external(connec_entity))
+  end
+
+  def update_entity_to_external(client, connec_entity, external_id)
+    data = data_to_external(connec_entity)
+    data['ID'] = external_id
+    client.update(self.external_entity_name, data)
+  end
+
+  def push_entities_to_connec(connec_client, external_entities, organization)
+    external_entities.each do |external_entity|
+      idmap = IdMap.find_or_create_by(salesforce_id: external_entity.Id, salesforce_entity: self.external_entity_name, organization_id: organization.id)
+      # Entity does not exist in Connec!
+      if idmap.connec_id.blank?
+        connec_entity = self.create_entity_to_connec(connec_client, external_entity)
+        idmap.update_attributes(connec_id: connec_entity['id'], connec_entity: self.connec_entity_name.downcase)
+      else
+        connec_entity = self.update_entity_to_connec(connec_client, external_entity, idmap.connec_id)
+      end
+    end
+  end
+
+  def create_entity_to_connec(client, external_entity)
+    response = client.post("/#{self.connec_entity_name.downcase.pluralize}", { "#{self.connec_entity_name.downcase.pluralize}": data_to_connec(external_entity) })
+    JSON.parse(response.body)["#{self.connec_entity_name.downcase.pluralize}"]
+  end
+
+  def update_entity_to_connec(client, external_entity, connec_id)
+    data = data_to_connec(external_entity)
+    data['id'] = connec_id
+    response = client.post("/#{self.connec_entity_name.downcase.pluralize}", { "#{self.connec_entity_name.downcase.pluralize}": data })
```

Comment:
To update a resource on Connec! API, you need to send a PUT request against the endpoint `resources/id`
For instance:
`PUT /organizations/connec_id`